src/bootstrap/src/core/config/toml/rust.rs

```diff
@@ -394,6 +393,23 @@ pub fn check_incompatible_options_for_ci_rustc(
     Ok(())
 }
 
+pub(crate) const VALID_CODEGEN_BACKENDS: &[&str] = &["llvm", "cranelift", "gcc"];
+
+pub(crate) fn validate_codegen_backends(
+    backends: Vec<String>,
+    section: impl FnOnce() -> String,
+) -> Vec<String> {
+    for backend in &backends {
+        if !VALID_CODEGEN_BACKENDS.contains(&&**backend) {
+            let section = section();
+            panic!(
+                "Invalid value '{backend}' for '{section}.codegen-backends'. Permitted values: {VALID_CODEGEN_BACKENDS:?}."
```

Comment:
mh, I remember a PR deliberately introduce the fallthrough to the previous `else` if this condition fails to obtain? it was about using an out-of-tree codegen backend, I believe.

cc @FractalFir @WaffleLapkin

Reply:
I suspect the intent was to panic for codegen backend names starting with `rustc_codegen_`, while only warning for unknown backends. As written before this PR however it would only panic/warn when the name starts with `rustc_codegen_` and don't check the name otherwise however.